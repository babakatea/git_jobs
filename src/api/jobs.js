import axios from "axios";

const url = 'http://localhost:3000/api/get_jobs/';

async function fetchJobs() {
    const {data} = await axios.get(url);

    return data;
}

export {fetchJobs};
