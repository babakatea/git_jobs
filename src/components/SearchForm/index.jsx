import React, {Component} from 'react';
import './index.less'

class SearchForm extends Component {
    render() {
        return (
            <div className="hero-image">
                <h1>The quickest way to  </h1>
                <h1>find a JOB</h1>

                <form>
                    <button type="submit">Get Started</button>
                </form>
            </div>
        );
    }
}

export default SearchForm;


// TODO:
//
// 1. Write upper the main text
// 2. Add ore space between button and text
// 3. Align to right a little
// 4. Menu text more bigger and color more attractive
// 5. Add logo
// 6. Write below photo the information about website
// 7. Add links
// 8. Connect to API
// 9. Search and esults page
// 10. Authorization
// 11. Profile page
// 12. Statistics page