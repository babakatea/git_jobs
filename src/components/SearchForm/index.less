.hero-image {
  text-align: left;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 50px;
  height: calc(100vh - 173px);
  background-image: url("images/4.jpg");
  background-repeat: no-repeat;
  background-size: 100% 100%;
  margin: 1px;

  h1 {
    color: #111;
    font-family: 'Lato', sans-serif;
    font-size: 55px;
    font-weight: 300;
    line-height: 64px;
    margin: 0 0 0;
    //padding: 5px 30px;
    padding-left: 5px;
  }


  form {
    button {
      font-size: 14px;
      border: 1px solid #dcdcdc;
      padding: 8px 30px;
      cursor: pointer;
      background: white;
      transition: 0.2s;
      border-radius: 25px;

      &:hover {
        background: #dcdcdc;
      }

    }
  }
}

