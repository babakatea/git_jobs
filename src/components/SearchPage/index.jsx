import React, {Component} from "react";

class SearchForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: [],
        }
    }


    componentWillMount() {
        fetch('https://jobs.github.com/positions.json?search=node', {
            mode: "cors", // no-cors, cors, *same-origin
            cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
            credentials: "same-origin", // include, *same-origin, omit
            headers: {
                "Content-Type": "application/json",
                "Access-Control-Allow-Origin": "*",
            },
        })
            .then(response => this.setState({data: response.data})) // parses response to JSON
            .catch(err => console.log(err));
    }

    render() {
        return (
            this.state.data.map(item => <div key={item.id}>{item.text}</div>)
        );
    }
}

export default SearchForm;
//https://jobs.github.com/positions.json?search=node