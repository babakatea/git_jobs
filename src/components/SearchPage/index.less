

.jobs-list {
  padding: 10px;
}

.jobs-entry {
  background: #f5f5f5;
  padding: 10px;
  margin: 10px;
}

.search-fields {
  width: 400px;
  margin-right: 15px;

}

.search-page {
  width: auto;
  overflow-x: hidden;

}

#search {
  margin-top: 20px;
  margin-left: 30px;
  height: 50px;

}

#name-input {

}