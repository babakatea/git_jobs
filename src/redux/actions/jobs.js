import services from '../services';
import {jobsActionTypes} from "../constants";

const success_list_jobs = (list) => ({
  type: jobsActionTypes.JOBS_LIST_LOADED,
  list
});

const fail_list_jobs = (error) => ({
  type: jobsActionTypes.JOBS_LIST_LOAD_FAILED, error
});

const success_details_job = (details) => ({
  type: jobsActionTypes.JOBS_DETAILS_LOADED,
  details
});

const fail_details_job = (error) => ({
  type: jobsActionTypes.JOBS_DETAILS_LOAD_FAILED,
  error
});

const request_list_jobs = () => ({
  type: jobsActionTypes.JOBS_LIST_REQUEST,
});

const request_job_details = () => ({
  type: jobsActionTypes.JOBS_DETAILS_LOAD_REQUESTED,
});

export const loadJobsList = (params) => {
  return dispatch => {
    dispatch(request_list_jobs());
    return services.getJobs(params).then(data => {
      dispatch(success_list_jobs(data));
    }, error => {
      dispatch(fail_list_jobs(error));
    });
  };
};

export const loadJobsDetails = (params) => {
  return dispatch => {
    dispatch(request_job_details());
    return services.getDetails(params).then(data => {
      dispatch(success_details_job(data));
    }, error => {
      dispatch(fail_details_job(error));
    });
  };
};
