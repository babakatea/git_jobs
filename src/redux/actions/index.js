import alerts from './alerts';
import users from './users';
import {loadJobsDetails, loadJobsList} from './jobs';

export default {alerts, users, jobs: {loadJobsDetails, loadJobsList}};
