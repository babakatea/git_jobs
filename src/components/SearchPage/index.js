import React from "react";
import './index.less';
import {grommet} from "grommet/themes";
import {connect} from "react-redux";
import {
    Box,
    Button,
    Grommet,
    Form,
    FormField,
    CheckBox,
} from "grommet";
import {loadJobsList} from "../../redux/actions/jobs";


class SearchForm extends React.Component {
    state = {
        checked: false,
        activePage: 1,
        pages: [
            {
                number: 1,
                active: true
            },
            {
                number: 2,
                active: false
            },
            {
                number: 3,
                active: false
            },
            {
                number: 4,
                active: false
            },
            {
                number: 5,
                active: false
            },
            {
                number: 6,
                active: false
            },
            {
                number: 7,
                active: false
            }
        ],
    };

    componentDidMount() {
        !this.props.list.length > 0 && this.props.dispatch(loadJobsList({method: 'GET'}));
    }

    buildDetailsClickHandler = (job) => () => {
        this.props.history.push(`/job/${job.id}`)
    };

    onPageChange = (event) => {
        let newPages = this.state.pages.map(page => ({
            number: page.number,
            active: page.number === parseInt(event.target.name)
        }));

        console.log(this.state.pages);

        this.setState({pages: newPages});
        const filters = {
            page: event.target.name,
        };

        this.props.dispatch(loadJobsList({method: 'GET', params: filters}));
    };

    onSubmit = (params) => {
        const filters = {
            description: params.value.description,
            location: params.value.location,
            full_time: params.value.full_time,
            page: 1,
        };

        this.props.dispatch(loadJobsList({method: 'GET', params: filters}));
    };

    render() {
        return (
            <div className="search-page">
                <div>
                    <Grommet theme={grommet}>
                        <Form onSubmit={this.onSubmit}>
                            <Box direction="row" pad="medium" justify="center">
                                <FormField id="name-input"
                                           placeholder="Filter by title, benefits, companies"
                                           className="search-fields" label="Job description" name="description">
                                </FormField>
                                <FormField id="location-input" className="search-fields" label="Location"
                                           name="location"
                                           placeholder="Filter by city, state, zip code or country">
                                </FormField>
                                <div className="check-box">
                                    <FormField
                                        id="full-time"
                                        name="full_time"
                                        component={CheckBox}
                                        pad={true}
                                        label="Full Time Only"
                                        className={"no-border"}
                                    />
                                </div>
                                <Button id="search" type="submit" label="Search" primary/>
                            </Box>
                        </Form>
                    </Grommet>
                </div>

                <div className={'jobs-list'}>
                    {this.props.list && this.props.list.map((job) =>
                        <div key={job.id} job={job} className={'jobs-entry'}>
                            <p onClick={this.buildDetailsClickHandler(job)} className="job-name">
                                <span>{job.title}</span>
                            </p>
                            <p>{job.type}</p>
                            <p className="job-location">{job.location}</p>
                            <button className="button-like">
                                <span>Like</span>
                            </button>

                        </div>
                    )}
                    <div className="pagination">
                        <a>&laquo;</a>
                        {this.state.pages.map(page => (
                            <a key={page.number} onClick={this.onPageChange} name={page.number}
                               className={page.active ? "active" : ""}>{page.number}</a>
                        ))}
                        <a>&raquo;</a>
                    </div>
                </div>
            </div>
        );
    }

}

const mapStateToProps = (state) => {
    const {jobs} = state;

    return jobs;
};

export default connect(mapStateToProps)(SearchForm);
