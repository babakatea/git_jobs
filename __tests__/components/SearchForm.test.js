import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import createStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {BrowserRouter as Router, Route} from 'react-router-dom';
import SearchPage from "../../src/components/SearchForm";
import SearchForm from "../../src/components/SearchForm";
import {mount, shallow} from "enzyme";
import {loadJobsList} from "../../src/redux/actions/jobs";
import {toJson} from "enzyme-to-json";


describe('List of jobs page', () => {
  const mockStore = createStore([thunk]);
  const initialState = {
    jobs: []
  };
  const store = mockStore(initialState);

  it('Test render component', (done) => {
    const container = renderer.create(
        <Provider store={store}>
          <Router>
            <Route component={SearchForm}/>
          </Router>
        </Provider>,
        {createNodeMock: ({type}) => document.createElement(type)}
      );

    expect(container.toJSON()).toMatchSnapshot();

    done();
  });

  it('Test render non-empty component', () => {
    const container = renderer.create(
      <Provider store={store}>
        <Router>
          <Route component={SearchForm}/>
        </Router>
      </Provider>,
      );

    expect(container.toJSON()).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it('successfully calls on like handle click function from props', () => {
    const mockClick = jest.fn();

    const mockStore = createStore([thunk]);
    const initialState = {
      jobs: []
    };
    const store = mockStore(initialState);

    const component = mount(
      <Provider store={store}>
        <SearchForm handleLikeClick={mockClick}/>
      </Provider>
    );


    component.find('button').simulate('click');
    expect(mockClick).toBeCalled();
  });
});


