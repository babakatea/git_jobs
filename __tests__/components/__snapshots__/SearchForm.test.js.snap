// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List of jobs page Test render component 1`] = `
<div
  className="search-page"
>
  <div>
    <div
      className="StyledGrommet-sc-19lkkz7-0 bYWoze"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 jhDpEr"
        >
          <div
            className="search-fields FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Job description
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cuRQxH"
                  id="name-input"
                  label="Job description"
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Filter by title, benefits, companies"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="search-fields FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Location
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cuRQxH"
                  id="location-input"
                  label="Location"
                  name="location"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Filter by city, state, zip code or country"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="check-box"
          >
            <div
              className="no-border FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 gZSidW"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 dJuAoq"
                >
                  <label
                    checked={false}
                    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 bVmNDQ"
                    htmlFor="full-time"
                    onClick={[Function]}
                  >
                    <div
                      checked={false}
                      className="StyledCheckBox-sc-1dbk5ju-6 kJHHiF StyledBox-sc-13pk1d4-0 emCRMK"
                    >
                      <input
                        checked={false}
                        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 brHjRm"
                        id="full-time"
                        name="full_time"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                      />
                      <div
                        checked={false}
                        className="StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 fxKnpz StyledBox-sc-13pk1d4-0 jFlQDV"
                      />
                    </div>
                    <span>
                      Full Time Only
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <button
            className="StyledButton-sc-323bzc-0 cawECJ"
            id="search"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="jobs-list"
  >
    <div
      className="pagination"
    >
      <a>
        «
      </a>
      <a
        className="active"
        name={1}
        onClick={[Function]}
      >
        1
      </a>
      <a
        className=""
        name={2}
        onClick={[Function]}
      >
        2
      </a>
      <a
        className=""
        name={3}
        onClick={[Function]}
      >
        3
      </a>
      <a
        className=""
        name={4}
        onClick={[Function]}
      >
        4
      </a>
      <a
        className=""
        name={5}
        onClick={[Function]}
      >
        5
      </a>
      <a
        className=""
        name={6}
        onClick={[Function]}
      >
        6
      </a>
      <a
        className=""
        name={7}
        onClick={[Function]}
      >
        7
      </a>
      <a>
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`List of jobs page Test render non-empty component 1`] = `
<div
  className="search-page"
>
  <div>
    <div
      className="StyledGrommet-sc-19lkkz7-0 bYWoze"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 jhDpEr"
        >
          <div
            className="search-fields FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Job description
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cuRQxH"
                  id="name-input"
                  label="Job description"
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Filter by title, benefits, companies"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="search-fields FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Location
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cuRQxH"
                  id="location-input"
                  label="Location"
                  name="location"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Filter by city, state, zip code or country"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="check-box"
          >
            <div
              className="no-border FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 gZSidW"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 dJuAoq"
                >
                  <label
                    checked={false}
                    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 bVmNDQ"
                    htmlFor="full-time"
                    onClick={[Function]}
                  >
                    <div
                      checked={false}
                      className="StyledCheckBox-sc-1dbk5ju-6 kJHHiF StyledBox-sc-13pk1d4-0 emCRMK"
                    >
                      <input
                        checked={false}
                        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 brHjRm"
                        id="full-time"
                        name="full_time"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="checkbox"
                      />
                      <div
                        checked={false}
                        className="StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 fxKnpz StyledBox-sc-13pk1d4-0 jFlQDV"
                      />
                    </div>
                    <span>
                      Full Time Only
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <button
            className="StyledButton-sc-323bzc-0 cawECJ"
            id="search"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="jobs-list"
  >
    <div
      className="pagination"
    >
      <a>
        «
      </a>
      <a
        className="active"
        name={1}
        onClick={[Function]}
      >
        1
      </a>
      <a
        className=""
        name={2}
        onClick={[Function]}
      >
        2
      </a>
      <a
        className=""
        name={3}
        onClick={[Function]}
      >
        3
      </a>
      <a
        className=""
        name={4}
        onClick={[Function]}
      >
        4
      </a>
      <a
        className=""
        name={5}
        onClick={[Function]}
      >
        5
      </a>
      <a
        className=""
        name={6}
        onClick={[Function]}
      >
        6
      </a>
      <a
        className=""
        name={7}
        onClick={[Function]}
      >
        7
      </a>
      <a>
        »
      </a>
    </div>
  </div>
</div>
`;
